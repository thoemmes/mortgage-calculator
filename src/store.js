import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';


// const ui = (state, action) => {
//     console.log(state);
//     switch (action.type) {
//         case 'SET_MORTGAGE_SLIDER':
//         case 'MORTGAGE_SLIDER_MOVING':
//             return {
//                 ...state,
//                 mortgageSlider: mortgageSlider(state.mortgageSlider, action)
//             };
//         case 'SET_AMORTISATION_SLIDER':
//         case 'AMORTISATION_SLIDER_MOVING': 
//             return{
//                 ...state,
//                 amortisationSlider: amortisationSlider(state.amortisationSlider, action)
//             }; 

//         default:
//             return { 
//                 mortgageSlider : mortgageSlider(state, action),
//                 amortisationSlider : amortisationSlider(state, action)
//             };
//     }
// }



const mortgageSlider = (state = {
    position: 0.5,
    isMoving: false
}, action) => {
    switch (action.type) {
        case 'SET_MORTGAGE_SLIDER':
            return {
                ...state,
                position: action.position,
            };
        case 'MORTGAGE_SLIDER_MOVING':
            return {
                ...state,
                isMoving: action.isMoving
            }
        default:
            return state;
    }
}

const amortisationSlider = (state = {
    position: 0.5,
    isMoving: false
}, action) => {
    switch (action.type) {
        case 'SET_AMORTISATION_SLIDER':
            return {
                ...state,
                position: action.position,
            };
        case 'AMORTISATION_SLIDER_MOVING':
            return {
                ...state,
                isMoving: action.isMoving
            }
        default:
            return state;
    }
}


const interestSlider = (state = {
    position: 0.5,
    isMoving: false
}, action) => {
    switch (action.type) {
        case 'SET_INTEREST_SLIDER':
            return {
                ...state,
                position: action.position,
            };
        case 'INTEREST_SLIDER_MOVING':
            return {
                ...state,
                isMoving: action.isMoving
            }
        default:
            return state;
    }
}


const ui = combineReducers({mortgageSlider, amortisationSlider, interestSlider});

const mortgageMax = (state = 500000
    , action) => {
    switch (action.type) {

        case 'SET_MORTGAGE_MAX': {
            return action.max;

        }
        default:
            return state;
    }
}

const amortisationMax = (state = 2000
    , action) => {
    switch (action.type) {

        case 'SET_AMORTISATION_MAX': {
            return action.max;

        }
        default:
            return state;
    }
}

const interestMax = (state = 5
    , action) => {
    switch (action.type) {

        case 'SET_INTEREST_MAX': {
            return action.max;

        }
        default:
            return state;
    }
}


// const interest = (state = 0.02, action) => {
//     switch (action.type) {
//         case 'SET_INTEREST':
//             return action.interest;
//         default:
//             return state;
//     }
// }

// const amortisation = (state = 0.02, action) => {
//     switch (action.type) {
//         case 'SET_AMORTISATION':
//             return action.amortisation;
//         default:
//             return state;
//     }
// }


export const reducers = combineReducers({
    ui,
    mortgageMax,
    //interest,
    amortisationMax,
    interestMax
});

const store = createStore(reducers, applyMiddleware(thunkMiddleware));


export default store;




// const mortgage = (mortgage: 50000, action) => {
//     switch (action.type) {
//         case 'MORTGAGE':
//             return {
//                 ...state,
//                 mortgage: action.amount
//             };
//     }

//     return state;
// }

// const mortgageSliderPosition = (state = 0.5, action) => {
//     switch (action.type) {
//         case 'SET_MORTGAGE_SLIDER':
//             return action.position;
//         default:
//             return state;
//     }
// }

// const mortgageSliderCursor = (state = 'default', action) => {
//     switch (action.type) {
//         case 'SET_CURSOR':
//             return action.cursor;
//         default:
//             return state;
//     }
// }

// const mortgageSliderMove = (state = 'default', action) => {
//     switch (action.type) {
//         case 'MORTGAGE_SLIDER_MOVE':
//             return action.move ? 'e-resize' : 'default';
//         default:
//             return state;
//     }
// }
