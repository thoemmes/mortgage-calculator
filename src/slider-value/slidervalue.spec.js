import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import SliderValue from './slidervalue';
import SliderMortgageComponent from './slidermortgagecomponent';

import { reducers } from '../store';
import { createStore } from 'redux';

describe('SliderValue', () => {

    it('renders SliderValue', () => {
        const sliderValue = renderer.create(<SliderValue sliderPosition="0.8" cursor="e-resize" unit="â‚¬" inputValue="50000" />);
        expect(sliderValue).toMatchSnapshot();
    });

    describe('SliderMortgageComponent', () => {

        let store;
        beforeEach(() => {
            store = createStore(reducers);

        });
       
        it('sets the slider to new position when inputValue is changed to a value lesser than max', () => {
            const sliderMortgageComponent = mount(<SliderMortgageComponent store={store} />);
            sliderMortgageComponent.find('input').simulate('change', { target: { value: 400000 } });
            let handle = sliderMortgageComponent.find('.outer_handle');
            expect(handle.prop('style').marginLeft).toEqual('80%');

        });

        
        it('sets the slider to new max when inputValue is changed to a value greater than max', () => {
            const sliderMortgageComponent = mount(<SliderMortgageComponent store={store} />);
            sliderMortgageComponent.find('input').simulate('change', { target: { value: 600000 } });
            let handle = sliderMortgageComponent.find('.outer_handle');
            expect(handle.prop('style').marginLeft).toEqual('100%');
            expect(store.getState().mortgageMax).toEqual(600000);

        });
    });

});
