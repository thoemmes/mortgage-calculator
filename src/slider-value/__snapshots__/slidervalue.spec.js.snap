// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderValue renders SliderValue 1`] = `
<div
  className="slidervalue"
>
  <div
    className="slider"
    onMouseDown={undefined}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={undefined}
    style={
      Object {
        "cursor": "e-resize",
      }
    }
  >
    <div
      className="outer_handle"
      draggable="false"
      style={
        Object {
          "cursor": "e-resize",
          "marginLeft": "80%",
        }
      }
    >
      <span
        className="handle"
      >
        IIII
      </span>
    </div>
  </div>
  <input
    className="value"
    onChange={[Function]}
    step={1000}
    type="number"
    value="50000"
  />
  <div
    className="unit"
  >
    â‚¬
  </div>
</div>
`;
